test/src/test/java/org/corfudb/runtime/view/AddressSpaceViewTest.java

```diff
@@ -78,6 +80,44 @@ public void ensureStripingWorks()
     }
 
     @Test
+    public void testCacheEvictionTimer() throws Exception {
+        addServer(SERVERS.PORT_0);
+
+        bootstrapAllServers(new TestLayoutBuilder()
+                .setEpoch(1L)
+                .addLayoutServer(SERVERS.PORT_0)
+                .addSequencer(SERVERS.PORT_0)
+                .buildSegment()
+                .buildStripe()
+                .addLogUnit(SERVERS.PORT_0)
+                .addToSegment()
+                .addToLayout()
+                .build());
+
+        int bufSize = 10;
+        int numEntriesInCache = 3;
+        long expiryCounter = 1;
+
+        CorfuRuntime rt  = new CorfuRuntime(getDefaultConfigurationString())
+                .setCacheExpiryTime(expiryCounter)
+                .setMaxCacheSize(bufSize * numEntriesInCache)
+                .connect();
+
+        byte[] data = new byte[bufSize];
+        ByteBuf buf = Unpooled.buffer().writeBytes(data);
+        ILogData entry = new LogData(DataType.DATA, buf);
+        int size = entry.getSizeEstimate();
+        for(long x = 0; x < 4; x++) {
+            rt.getAddressSpaceView().readCache.put(x, entry);
+        }
+
+        Thread.sleep(2000);
```

Comment:
Use [FakeTicker](https://github.com/ben-manes/caffeine/wiki/Testing) to control time.